src/helpers/create-project-card.ts

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2021 Expedia, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { getDestinationColumn, getProjectColumns } from '../utils/get-project-columns';
+import { PullRequest } from '../types';
+import { context } from '@actions/github';
+import { octokit } from '../octokit';
+
+interface CreateProjectCardProps {
+  teams?: string;
+  pull_number: number;
+  login?: string;
+  project_name: string;
+  project_destination_column_name: string;
+  note?: string;
+}
+
+export const createProjectCard = async ({ pull_number, project_name, project_destination_column_name, note }: CreateProjectCardProps) => {
+  const getResponse = await octokit.pulls.get({ pull_number, ...context.repo });
+  const pullRequest = getResponse.data as PullRequest;
+  const columnsList = await getProjectColumns({ project_name });
+
+  if (!columnsList) {
+    return null;
+  }
+
+  const destinationColumn = getDestinationColumn(columnsList, project_destination_column_name);
+  const cardParams = generateCardParams(note, destinationColumn, pullRequest);
+  if (destinationColumn) {
+    return octokit.projects.createCard(cardParams);
+  }
+};
+
+const generateCardParams = (note: string | undefined, filteredColumn: any, pullRequest: PullRequest) => {
+  if (note) {
+    return {
+      column_id: filteredColumn?.id,
+      note,
+      ...context.repo
+    };
+  }
+
+  return {
+    column_id: filteredColumn?.id,
+    content_id: pullRequest.id,
+    content_type: 'PullRequest',
```

Comment:
Why do we want to include these params only if there is no "note"?

Reply:
![Screen Shot 2021-12-08 at 11 34 02 AM](https://user-images.githubusercontent.com/40922393/145255881-9e60ee63-0a8b-4f94-af04-3dc77149a520.png)
According to the [docs](https://octokit.github.io/rest.js/v18), it seems that all of these params are required for `createCard()`

Reply:
Not sure why the docs says that they are required, but if we send either `content_id` or `content_type` or both when we are also sending a note, it throws an error saying the following:
![Screen Shot 2021-12-08 at 12 00 09](https://user-images.githubusercontent.com/30817795/145259820-8304f0e4-53e4-4fb1-91d1-71a583eac7cd.png)